src/components/Button/components/ButtonIcon/index.tsx

```diff
@@ -40,6 +42,10 @@ export default (props: ButtonIconProps) => {
         case 'refresh':
             return <RefreshOutlinedIcon fontSize="small" />;
         default:
+            if (Object.keys(icons).includes(props.type as string)) {
+                return <Icon name={props.type as IconName} fontSize="small" />;
+            }
+
             return <SettingsOutlinedIcon fontSize="small" />;
```

Comment:
It'd be a bit neater to add a `case 'settings':` and then have the default create the Material UI Icon. There shouldn't be a need to check Object.keys that way as we're protected by the type :)

Reply:
yes, but this would be a breaking change as the current implementation expects the settings icon to be the default case (there are tests that cover it as well). 

I can update this and the test, by any consumer that uses this would face an issue if they've been using it considering this behaviour

Reply:
Unless I'm mistaken this wouldn't be a breaking change as the `icon` prop must be one of the existing strings (`'add_plus' | 'copy' | 'external' | 'folder' | 'refresh' | 'settings'`). If the `icon` prop isn't specified, no icon is rendered ([see here](https://github.com/aws/aws-northstar/blob/main/src/components/Button/index.tsx#L104-L107)).

I had a quick look at the button tests and but couldn't find one that checks that the `settings` icon is shown by default at all - I wonder if you could point me to which one? :)

Reply:
The test is here: 

https://github.com/aws/aws-northstar/blob/main/src/components/Button/components/ButtonIcon/index.test.tsx

And `ButtonIcon` is exported as well so could be used directly without using the Button

https://github.com/aws/aws-northstar/blob/main/src/components/index.ts#L20

Reply:
also the `type` prop for `ButtonIcon` is optional as well: https://github.com/aws/aws-northstar/blob/main/src/components/Button/components/ButtonIcon/index.tsx#L26-L28

Reply:
This test doesn't expect that the icon is a `settings` icon - it simply checks that an `svg` is rendered, so it shouldn't be broken by this change :)

I hadn't realised the `ButtonIcon` was publicly exposed, so I can see that it wouldn't be backwards compatible for those using the `ButtonIcon` component directly, but would be backwards compatible for those using the `Button` component.

I'm not sure our users would use the ButtonIcon directly, and rely on the behaviour that `settings` is the default icon though - what do you think @jessieweiyi ?

Reply:
Yes it doesn't expect that is a `settings` icon, but it expect that there's an icon. If a consumer uses it as such and suddenly a new version instead of having a `settings` icon renders nothing (or a different one), it's still a breaking change. 

I don't mind to make the change but I'm just highlighting that modifying it in that way will results (potentially) in unexpected behaviours. Also worth to mention that type-check while it helps will still not save from wrong usage of the API. I got to know about the fact that was rendering a `settings` icon by default by passing random values to the `icon` prop (trying to see whether could use also other Material UI icons, as I'm not using typescript in the project I'm working on types are not enforced)

Reply:
Also the way you mention is how I was working on the change initially but I figured out that would break the test and the API. But again, if you guys are ok with this (as it's an edge case anyway) I'll make the change and fix the test

Reply:
Gotcha, that makes sense 👍 

My feeling is the `type` for `ButtonIcon` should be required and we should accept that it's not backwards compatible for those using the `ButtonIcon` component directly, without specifying a type. Showing a settings icon feels to me like counterintuitive default behaviour for a publicly exposed component. Keen to hear Jessie's opinion on this too though :)

I don't want this to block you though - let's leave it as-is for now, and if you could add a simple test that an icon is rendered when you pass a Material UI icon name I'm happy to approve :)

Reply:
agree on that, definitely better to use your approach (as it was they way I wanted implement it initially). Let me know if that requires changes I'm happy to reiterate on it change it accordingly.

I've updated the PR to include an additional test

Reply:
Thanks! :)